pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -253,14 +252,6 @@ func (r *Reconciler) handleFinalizers(ctx context.Context, app *v1beta1.Applicat
 			klog.InfoS("Register new finalizer for application", "application", klog.KObj(app), "finalizer", resourceTrackerFinalizer)
 			return true, errors.Wrap(r.Client.Update(ctx, app), errUpdateApplicationFinalizer)
 		}
-		if appWillRollout(app) {
```

Comment:
why this removed

Reply:
Before, we use `list` to delete resource trackers only when it's created by app rollout. Now, we  use `list` to delete any resource trackers. So it's no need to handle approllout-created ones.